pkg/vmtest/integration.go

```diff
@@ -75,6 +75,21 @@ type Options struct {
 	// collected and saved to:
 	//   u-root/integration/coverage/{{testname}}/{{instance}}/kernel_coverage.tar
 	NoKernelCoverage bool
+
+	// AdditionalTestDataFiles collects an additional list testdata files to be
+	// placed into the temporary directory, exposed to the VM.
+	//
+	// Currently those files are relative to test data root /testdata/ from within
+	// a test VM.
+	AdditionalTestDataFiles []TestDataFile
```

Comment:
this argument is not necessary -- you can create a file in the test setup, and add it to the VM using BuildOpts.ExtraFiles (just like -extra-files on the u-root command line)

Reply:
This file does not have to be part of initramfs cpio. Bootstrapping it via testdata tmpdir seems faster, and this option helps bootstrap many small files / configs quicker than creating them one by one, and write to them, and specify via BuildOpts

Reply:
It's not an API we will support in https://github.com/hugelgupf/vmtest when it replaces vmtest here in u-root. I'd like there to be not more than 2 mechanisms to share files with the VM.

You could also use t.TempDir() in the test, create the file, and pass t.TempDir() as Options.TmpDir, and it'll show up in the VM, the other mechanism that already exists.

Reply:
any reason it moves to separate repo ? -- that always make things harder to maintain

Reply:
The new repo is an experiment for now, to iterate on a new API.

I do like it though, because it makes the infra generally reusable by other people, and not a u-root-specific thing.

Reply:
what are the general hurdles to re-use it when same pieces are part of u-root ? -- it is pkg in u-root anyway 

and how many others are going to use the separate repo ?

Reply:
We're... off topic now. I'd rather not introduce a 3rd mechanism to share files with the VM. Please use one of the 2 existing ones.

For the off-topic conversation: pkg/vmtest inside u-root is not discoverable for other people in the Go ecosystem. I'd like it to be a framework that other people use. Only low complexity stuff from u-root is reused by other people at the moment, because that's somewhat discoverable and then easy to figure out. "It's hard to deal with in our internal infrastructure" is not a reason to not do something externally.

Reply:
Done.

I still don't get the point. The preference to not introduce a new way to add test file came across, but why is that and why the particular new way introduced here is a no-go did not come across, and there reads no explanation on that from previous comments. To me, those are closer to "policy statements". I see upsides of clear policies. I just don't get why.

On benefits, the proposed way is way more concise to bootstrap files under selected temporary directory, exposed to VM. 

From my view, a new way to do same thing is good, as long as it is net positive, and not introducing conflicting scenarios with existing ones. As a user, I prefer a config option over writing up 10lines of code to do the same thing in each test function.

Removed original proposal here, and switched to do it via a `t.TempDir`. That brings in additional lines of code just for this single test function. But I hope you do re-consider introducing similar AdditionalTestdataFile options in your separate repo.

Reply:
More API means more to test. One of the reasons I started experimenting and iterating in the other repo is that the current vmtest framework is (actually) horribly broken, like https://github.com/u-root/u-root/pull/2589 showed. I want to get the framework well-tested and under control before we start adding 3 different ways to do the same thing (and have to test 3 different ways to do the same thing).

You are right that if it adds value to the user, it should be considered. But I'm not for proliferating this thing before we fix what's here.

Reply:
Can you upload new revision?

Reply:
Done!